perun-openapi/openapi.yml

```diff
@@ -5613,122 +5848,797 @@ paths:
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/cabinetManager/findSimilarPublications/isbn:
+    get:
+      tags:
+        - CabinetManager
+      operationId: findSimilarPublicationsIsbn
+      summary: Finds similar publications
+      parameters:
+        - $ref: '#/components/parameters/isbn'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfPublicationForGUIResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/cabinetManager/findSimilarPublications/isbn-doi:
+    get:
+      tags:
+        - CabinetManager
+      operationId: findSimilarPublicationsIsbnDoi
+      summary: Finds similar publications
+      parameters:
+        - $ref: '#/components/parameters/isbn'
+        - $ref: '#/components/parameters/doi'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfPublicationForGUIResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/cabinetManager/findSimilarPublications/title:
+    get:
+      tags:
+        - CabinetManager
+      operationId: findSimilarPublicationsTitle
+      summary: Finds similar publications
+      parameters:
+        - $ref: '#/components/parameters/title'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfPublicationForGUIResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/cabinetManager/findSimilarPublications/title-doi:
+    get:
+      tags:
+        - CabinetManager
+      operationId: findSimilarPublicationsTitleDoi
+      summary: Finds similar publications
+      parameters:
+        - $ref: '#/components/parameters/title'
+        - $ref: '#/components/parameters/doi'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfPublicationForGUIResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/cabinetManager/findSimilarPublications/doi:
+    get:
+      tags:
+        - CabinetManager
+      operationId: findSimilarPublicationsDoi
+      summary: Finds similar publications
+      parameters:
+        - $ref: '#/components/parameters/doi'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfPublicationForGUIResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/cabinetManager/findSimilarPublications/title-isbn-doi:
+    get:
+      tags:
+        - CabinetManager
+      operationId: findSimilarPublicationsTitleIsbnDoi
+      summary: Finds similar publications
+      parameters:
+        - $ref: '#/components/parameters/title'
+        - $ref: '#/components/parameters/isbn'
+        - $ref: '#/components/parameters/doi'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfPublicationForGUIResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
```

Comment:
I think that instead of these methods, you can define only one with non-required parameters.